Fix Audit-ID header key

CaoShuFeng opened the pull request:
Now http header key "Audit-ID" doesn't have effect, because golang
automaticly transforms "Audit-ID" into "Audit-Id". This change use
http.Header.Get() function to canonicalize "Audit-ID" to "Audit-Id".


**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```
NONE
```

CaoShuFeng commented:
@sttts 
@timstclair

CaoShuFeng commented:
Before this change the following Audit-ID will not work:
curl  -H "Audit-ID: this-is-my-invalid-id" http://localhost:8080/version

sttts commented on staging/src/k8s.io/apiserver/pkg/audit/request.go:
```diff
@@ -51,6 +51,11 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
+
+	// golang canonicalizes the header name when adding values to the header map
+	// see https://golang.org/src/net/http/request.go#L150 for reference
+	ids = append(ids, req.Header[http.CanonicalHeaderKey(auditinternal.HeaderAuditID)]...)
```
ouch, I hate magic. Great finding!

sttts replied:
Can you link with a commit sha1? The referenced line will be invalid soon otherwise.

sttts replied:
I wonder, shouldn't we switch `auditinternal.HeaderAuditID` to the canonical form?

CaoShuFeng commented:
> I wonder, shouldn't we switch auditinternal.HeaderAuditID to the canonical form?
Done.

sttts commented:
/ok-to-test

CaoShuFeng commented:
/test pull-kubernetes-e2e-gce-etcd3

timstclair commented:
I think a better approach would be to change the usage:
```go
ids := req.Header[auditinternal.HeaderAuditID]
```
to
```go
ids := req.Header.Get(auditinternal.HeaderAuditID)
```
I think this is more idiomatic, and makes the constant definition independent of the internal go implementation.

CaoShuFeng committed:
Fix Audit-ID header key

Now http header key "Audit-ID" doesn't have effect, because golang
automaticly transforms "Audit-ID" into "Audit-Id". This change use
http.Header.Get() function to canonicalize "Audit-ID" to "Audit-Id".

CaoShuFeng commented:
@timstclair
Done.
Thanks for your suggestion.

sttts commented:
/approve no-issue
/lgtm

CaoShuFeng commented:
/test pull-kubernetes-unit

CaoShuFeng commented:
I wrote a test for this pr:
https://github.com/kubernetes/kubernetes/pull/48693
